### PR TITLE
doc: init readme

### PR DESCRIPTION
**Pull request by xiaoling-yi**
In this PR, we update the documentation!

The detailed microarchitecture graph will be added later. I'm just not good at drawing!

For software people, it should be enough without the microarchitecture graph. If we need more documentation, please let me know!

### CODE REVIEW

File: LICENSE
Comment (JosseVanDelm) [inline LICENSE:1]: I'm not a legal expert, but I think Apache is generally regarded as a software license, and not as a hardware license. You might want to look at http://solderpad.org/licenses/ or https://cern-ohl.web.cern.ch/
Snitch seems to use Apache 2.0 for sw, and SHL 0.51 for hardware

File: LICENSE [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+      To apply the Apache License to your work, attach the following
+      boilerplate notice, with the fields enclosed by brackets "[]"
+      replaced with your own identifying information. (Don't include
+      the brackets!)  The text should be enclosed in the appropriate
+      comment syntax for the file format. We also recommend that a
+      file or class name and description of purpose be included on the
+      same "printed page" as the copyright notice for easier
+      identification within third-party archives.
+
+   Copyright [yyyy] [name of copyright owner]
```
Comment (JosseVanDelm) [inline LICENSE:189]: Regardless of the license type, this field should be filled in

File: README.md [hunk 2]
```diff
@@ -131,129 +148,191 @@ The illustration of ldA, ldB, ldC, strideA, strideB, and strideC is listed below
...
 #### Programming model
+
 ```
 <T>gemmStridedBatched(
-int M, int N, int K,
-const T* A, int ldA, int strideA, 
-const T* B, int ldB, int strideB, 
-const T* C, int ldC, int strideC,
-int B)
+int Bacth, int M, int N, int K,
```
Comment (JosseVanDelm) [inline README.md:197]:
```suggestion
int Batch, int M, int N, int K,
```

**Review by JosseVanDelm — CHANGES_REQUESTED** [review]
Hi @xiaoling-yi I overall really like this PR, but I'm afraid Apache 2.0 might not be the right license, can we discuss this later?

**Review by guilherme-paim — APPROVED** [review]
Ok ! Nice Xiaoling !!!

**Pull request merged**